plugins/labormanager/joblabormapper.cpp

```diff
@@ -313,14 +311,27 @@ class jlfunc_construct_bld : public jlfunc
                 return workshop_build_labor[ws->type];
         }
         break;
+        case df::building_type::Furnace:
+        {
+            df::building_furnacest* frn = (df::building_furnacest*) bld;
+            if (frn->design && !frn->design->flags.bits.designed)
+                return df::unit_labor::ARCHITECT;
+            if (frn->type == df::furnace_type::Custom)
+            {
+                df::building_def* def = df::building_def::find(frn->custom_type);
+                return def->build_labors[0];
+            }
+            else
+                // cast to building_actual should be safe here because at this point the building has been designed
+                return construction_build_labor((df::building_actual*)bld);
```

Comment:
Also, building_furnacest inherits from building_actual, so you could probably pass in `frn` here without needing the cast